Library/Homebrew/rubocops/cask/stanza_order.rb

```diff
@@ -13,34 +13,48 @@ class StanzaOrder < Base
         extend AutoCorrector
         include CaskHelp
 
+        ON_SYSTEM_METHODS = RuboCop::Cask::Constants::ON_SYSTEM_METHODS
         MESSAGE = "`%<stanza>s` stanza out of order"
 
         def on_cask(cask_block)
           @cask_block = cask_block
-          add_offenses
+          add_offenses(toplevel_stanzas)
+
+          return unless (on_blocks = toplevel_stanzas.select { |s| ON_SYSTEM_METHODS.include?(s.stanza_name) }).any?
+
+          on_blocks.map(&:method_node).each do |on_block|
```

Comment:
We're doing this sort of thing in a bunch of cops, almost identically, so it should probably be moved somewhere central.